test/unit/CodeGenerator/Definitions/ResponseDtoDefinitionTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Unit\CodeGenerator\Definitions;
+
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\PropertyDefinition;
+use OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\ResponseDtoDefinition;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Property;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\CodeGenerator\Definitions\ResponseDtoDefinition
+ */
+final class ResponseDtoDefinitionTest extends TestCase
+{
+    /**
+     * @return mixed[]
+     */
+    public function responseDtoDefinitionProvider(): array
+    {
+        return [
+            [
+                'conditions' => ['hasProperties' => false],
+                'expected' => ['isEmpty' => true],
+            ],
+            [
+                'conditions' => ['hasProperties' => true],
+                'expected' => ['isEmpty' => false],
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed[] $conditions
+     * @param mixed[] $expected
+     *
+     * @dataProvider responseDtoDefinitionProvider
+     */
+    public function testResponseDtoDefinition(array $conditions, array $expected): void
```

Comment:
Only getStatusCode needs to be tested, all other methods of the class are inherited and final.

Reply:
Also we need to check the dispatch of parent properties in the constructor.